test/unit_tests/aws/wrapper/test_aws_client.py

```diff
@@ -0,0 +1,99 @@
+import dataclasses
+from typing import Union
+from unittest.mock import MagicMock, Mock, call, create_autospec
+
+from boto3 import Session
+
+from exasol_script_languages_container_ci_setup.lib.aws.wrapper.aws_client import AwsClient
+from test.mock_cast import mock_cast
+
+
+def test_init():
+    boto_session_factory = Mock()
+    aws_client = AwsClient(profile="profile", region="region",
+                           boto_session_factory=boto_session_factory)
+    assert boto_session_factory.mock_calls == []
```

Comment:
```suggestion
    expected = []
    boto_session_factory = Mock()
    aws_client = AwsClient(profile="profile", region="region",
                           boto_session_factory=boto_session_factory)
    actual = boto_session_factory.mock_calls
    assert actual == expected

```

Reply:
looks for me, this makes the test harder to read